src/Compilers/CSharp/Portable/Compiler/MethodBodySynthesizer.Lowered.cs

```diff
@@ -277,4 +285,342 @@ internal static BoundBlock ConstructSingleInvocationMethodBody(SyntheticBoundNod
+                    F.Assignment(F.Local(l), F.Convert(_long, F.Parameter(o))),
+                    F.Assignment(F.Parameter(value), F.Convert(_ulong, F.Local(l))),
+                    F.Return(F.Binary(BinaryOperatorKind.LongGreaterThan, _bool, F.Local(l), F.Literal((long)int.MinValue)))
+                    )),
+                //    if (t == typeof(ulong))
+                //    {
+                //        value = (ulong)o;
+                //        return (value > int.MaxValue);
+                //    }
+                F.If(F.ObjectEqual(F.Local(t), F.Typeof(_ulong)), F.Block(
+                    F.Assignment(F.Parameter(value), F.Convert(_ulong, F.Parameter(o))),
+                    F.Return(F.Binary(BinaryOperatorKind.ULongGreaterThan, _bool, F.Parameter(value), F.Literal((ulong)int.MaxValue)))
+                    )),
+                //    return false;
+                F.Return(F.Literal(false))
+                );
+
+            // NOTE: we created this block in its most-lowered form, so analysis is unnecessary
+            F.CloseMethod(body);
+        }
+    }
+
+    internal class SynthesizedAsLargeNegativeMethod : SynthesizedGlobalMethodSymbol
+    {
+        private readonly Emit.PEModuleBuilder _module;
+
+        public SynthesizedAsLargeNegativeMethod(Emit.PEModuleBuilder module, PrivateImplementationDetails pi, DiagnosticBag diagnostics, CSharpSyntaxNode node)
+            : base(module.SourceModule, pi, module.Compilation.GetSpecialType(SpecialType.System_Boolean), PrivateImplementationDetails.AsLargeNegativeName)
+        {
+            this._module = module;
+            var compilation = module.Compilation;
+            this.SetParameters(ImmutableArray.Create<ParameterSymbol>(
+                new SynthesizedParameterSymbol(this, compilation.GetSpecialType(SpecialType.System_Object), 0, RefKind.None, "o"),
+                new SynthesizedParameterSymbol(this, compilation.GetSpecialType(SpecialType.System_Int64), 1, RefKind.Out, "value")
+                ));
+            Binder.ReportUseSiteDiagnostics(this.ReturnType, diagnostics, node);
+            Binder.ReportUseSiteDiagnostics(this.Parameters[0].Type, diagnostics, node);
+            Binder.ReportUseSiteDiagnostics(this.Parameters[1].Type, diagnostics, node);
+        }
+
+        internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+        {
+            ///// <summary>
+            ///// If o is of an integral numeric type (namely, long) and contains a value that is less than
+            ///// any value of int, return true and set the value parameter to that value.  Otherwise it
+            ///// leaves an arbitrary value in value and returns false.
+            ///// </summary>
+            //internal static bool AsLargeNegative(object o, out long value)
+            //{
+            //    if (o == null || o.GetType() != typeof(long))
+            //    {
```

Comment:
You would avoid creating runtime type object.   isinst with a primitive type should be very cheap (null check + compare of klass/vtable pointer)

Sometimes JIT can optimize  "o.GetType() == typeof(...)", but not always so  isinst would seem to be preferable.
Anyways, it seems this entire approach is going to be shelved anyways.